crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -639,3 +655,301 @@ fn normalize_string(
 
     (normalized, newlines)
 }
+
+/// For docstring indentation, black counts spaces as 1 and tabs by increasing the indentation up
+/// to the next multiple of 8. This is effectively a port of `str.expandtabs`, which black calls.
+fn count_indentation_like_black(line: &str) -> TextSize {
+    let tab_width: u32 = 8;
```

Comment:
Should we use the tab size from the options when `IndentStyle::Tab` is used (we can fallback to 8 if it is spaces)?

Reply:
black always uses 8: https://github.com/psf/black/blob/c36e468794f9256d5e922c399240d49782ba04f1/src/black/strings.py#L61 . imo indentation that mixes tabs and space is fundamentally undefined, so i would pick something that's easy and stable or in this case, black compatible.

Reply:
That's correct, but black doesn't support changing the indent style.

I recommend running some doctest with the default options and once with indent style tab(4)

Reply:
do we actually allow setting the tab width somewhere (or did you mean something else)? `IndentStyle` is `enum IndentStyle { Tab, Space(u8) }`

Reply:
Oh we don't. Its an option on `PrinterOptions` but it seems we never initialize it: 

https://github.com/astral-sh/ruff/blob/3ef1c2e3030028178c87314d7528ecec074cc840/crates/ruff_formatter/src/printer/printer_options/mod.rs#L6-L7

Anyway. adding a test with indent style tab would be good. We can expose the tab with later.

Reply:
added 4 spaces, 2 spaces and tab fixture